### ISSUE TITLE
Allow setting entry properties by name 

### ISSUE THREAD

**Issue by stefanuebe** [stefanuebe/vaadin_fullcalendar#112]
Currently the set of entry properties is fixed on the server side and we have no way to simply override others then by extending the Entry class and overriding `toJson()`. This is not a very appealing way, especially when the client side properties are extended or we simply forgot some (e.g. text color).

Therefore the api should be extended to have some simple kind of "set(String key, Object value)" method, that stores the values in some internal map and uses them when applying "toJson".

Setter and getter should use that internal map, too. Any raw overriding of properties with checks is without guarantee and in the responsibility of the respective developer. 

Client side "update from server" needs to be extended to not only update a fixed set of properties, but any one.

**Comment by stefanuebe** [comment]
A sample workaround until this enhancement is implemented:

```
public static class ExtendedEntry extends ResourceEntry {
    @Override
    protected JsonObject toJson() {
        JsonObject jsonObject = super.toJson();
        jsonObject.put("textColor", "red"); // the "red" should come from some internal object field
        return jsonObject;
    }
}
```

**Comment by 961163289** [comment]
Can I use this method to set the font size?

**Comment by 961163289** [comment]
How to find the corresponding key?

**Comment by stefanuebe** [comment]
Fixed with version 3.2.0.
